### PR TITLE
feat(shape_estimation): add bicycle shape corrector

### PR DESCRIPTION
**Pull request by kaancolak**
## Description

Solves https://github.com/autowarefoundation/autoware.universe/issues/5859

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/shape_estimation/include/shape_estimation/corrector/bicycle_corrector.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public:
+  explicit BicycleCorrector(const bool use_reference_yaw = false)
+  : VehicleCorrector(use_reference_yaw)
+  {
+    corrector_utils::CorrectionBBParameters params;
+    params.min_width = 0.3;
+    params.max_width = 1.2;
+    params.default_width = (params.min_width + params.max_width) * 0.5;
+    params.min_length = 1.0;
+    params.max_length = 2.6;
```
Comment (YoshiRi) [inline perception/shape_estimation/include/shape_estimation/corrector/bicycle_corrector.hpp:32]: Can you tell me how did you decide this max length?
I searched and found longer motorcycle with 2.65m from [HONDA](https://www.honda.co.jp/VT1300/VT1300CR/size/).

How about expanding this a little bit?
```suggestion
    params.max_length = 2.8;
```

**Review by YoshiRi — CHANGES_REQUESTED** [review]
Great! Do you have any source how to decide this maximum size?

File: perception/shape_estimation/include/shape_estimation/corrector/bicycle_corrector.hpp (hunk 1 above)
Comment (kaancolak) [reply perception/shape_estimation/include/shape_estimation/corrector/bicycle_corrector.hpp:32]: Thank you so much @YoshiRi -san, some BossHoss models are around 2.7 m, and tandem bikes could be bigger but I don't have a reference. 

We can take your link as a reference.

**Review by YoshiRi — APPROVED** [review]
LGTM

**Pull request merged**